sos/report/plugins/__init__.py

```diff
@@ -1616,6 +1616,19 @@ def get_tags_for_file(self, fname):
                 tags.extend(val)
         return tags
 
+    def get_devices_by_fstype(self, fstype):
+        """Get list of devices based on fstype and returns empty array in case
+        of failure.
+        """
+        dev = []
+        all_devs = self.exec_cmd("lsblk -rpo NAME,FSTYPE")
+        if (all_devs['status'] == 0 and not all_devs['truncated']):
+            if all_devs['status'] == 0:
+                for line in all_devs['output'].splitlines():
+                    if (fstype in line) and (line.split()[0] not in dev):
+                        dev.append(line.split()[0])
+        return dev
```

Comment:
I'm torn about having this move from `Plugin` to `SoSReport`, and have it run alongside our device introspection, like for [block devs](https://github.com/sosreport/sos/blob/085ae37e0075a79e0fa16d93d011bc0b438dd0dc/sos/report/__init__.py#L505). Granted, these aren't _hardware devices_, but it's in the same vein as what we do for storage and network interfaces.

That way we run this once, when sos first initializes, and then in `Plugin` this function would instead simply filter through an already-made list/dict/whatever for those matching the requested `fstype`.

Reply:
That's a good point. I will into it, the main idea here is to reuse the info, not just for FS but also md.py can benefit from it as the FSTYPE in limited to FS as per `blkid -k`

Reply:
Sorry I should have been clearer - my intent was to suggest that this be at the same level as `storage`, `network`, or similar. So plugins would access it via something like `for fs in self.devices['filesystems']['xfs']: ...blah...`.

Again, this is a bit of a misnomer placing it under `devices`, so if you prefer we can also make a separate attribute to hold these, so that plugins could do something like `for fs in self.filesystems['xfs']:`